vgmdb_importer.user.js

```diff
@@ -0,0 +1,263 @@
+ *     Retailer Bonus, Event Only, Promo/Gift/Reward, Rental (General does not appear in API)
+ */
+function mapStatus(publishFormat) {
+    if (publishFormat.includes('Bootleg')) {
+        // Overrides promo
+        return 'bootleg';
+    } else if (publishFormat.includes('Promo/Gift/Reward')) {
+        return 'promotion';
+    } else {
+        return 'official';
+    }
+}
+
+/*
+ * Returns year, month and day dict based on ISO 8601 date string.
+ */
+function mapDate(releaseDate) {
+    const d = new Date(releaseDate);
+
+    return { year: d.getFullYear(), month: d.getMonth() + 1, day: d.getDay() };
+}
+
+/*
+ * Returns a likely label based on a list of VGMdb organizations.
+ */
+function mapLabel(organizations) {
+    const labelOrganization = getLabelOrganization(organizations);
+    if (labelOrganization) {
+        return labelOrganization['names']['en'];
+    } else if (organizations.length == 1) {
+        // If only one, assume that's the one
+        return organizations[0]['names']['en'];
+    } else {
+        return null;
+    }
+}
+
+/*
+ * Returns a list of MusicBrainz URLs based on the websites, stores and
+ * vgmdb_link values of the VGMdb API.
+ */
+function mapUrls(vgmdbLink, stores, websites) {
+    const urls = [];
+
+    urls.push({ url: vgmdbLink, link_type: 86 });
+
+    if (stores) {
+        for (const store of stores) {
+            // Filter out links to internal marketplace
+            if (store['link'].startsWith('http')) {
+                // Assumes mail order
+                urls.push({ url: store['link'], link_type: MBImport.URL_TYPES.purchase_for_mail_order });
+            }
+        }
+    }
+
+    if (websites) {
+        for (const commercial in websites['Commercial']) {
+            // Seems to fill same purpose as stores for albums
+            urls.push({ url: commercial['link'], link_type: MBImport.URL_TYPES.purchase_for_mail_order });
+        }
+    }
+
+    return urls;
+}
+
+/*
+ * Returns a list of disc objects each featuring title, format and tracklist
+ * based on VGMdb disc data and media format.
+ */
+function mapDiscs(vgmdbDiscs, mediaFormat) {
+    const discs = [];
+
+    let multipleMedia;
+    let generalMediaFormat;
+    if (mediaFormat.includes('+')) {
+        multipleMedia = true;
+    } else {
+        generalMediaFormat = extractVgmdbMedia(mediaFormat);
+    }
+
+    for (const vgmdbDisc of vgmdbDiscs) {
+        const disc = {};
+
+        if (multipleMedia) {
+            const discMediaFormat = extractVgmdbMedia(vgmdbDisc['name']);
+            disc['format'] = mapFormat(discMediaFormat);
+        } else {
+            disc['format'] = mapFormat(generalMediaFormat);
+        }
+
+        disc['tracks'] = mapTracks(vgmdbDisc['tracks']);
+
+        discs.push(disc);
+    }
+
+    return discs;
+}
+
+/*
+ * Returns a MusicBrainz format based on a VGMdb media format.
+ */
+function mapFormat(mediaFormat) {
+    return mediaFormat;
```

Comment:
It seems that at least `Digital` should be converted to `Digital Media`

Reply:
Whoops, meant to fill that in later on. Thanks!